backend/functions/views.py

```diff
@@ -1,30 +1,178 @@
+    get_batch_translations_using_indictrans_nmt_api,
+    get_batch_translations_using_google_translate,
+    check_translation_function_inputs,
+)
+
+
+## Utility functions
+def save_translation_pairs(
+    languages,
+    input_sentences,
+    output_dataset_instance_id,
+    batch_size,
+    api_type="indic-trans",
+):
+    """Function to save the translation pairs in the database.
+
+    Args:
+        languages (list): List of output languages for the translations.
+        input_sentences (list(list)): List of input sentences in list format containing the - [input_sentence, input_language, context, quality_status]
+        output_dataset_instance_id (int): ID of the output dataset instance.
+        batch_size (int): Number of sentences to be translated in a single batch.
+        api_type (str): Type of API to be used for translation. (default: indic-trans)
+            Allowed - [indic-trans, google]
+    """
+
+    # Iterate through the languages
+    for output_language in languages:
+
+        # Convert the input_sentences list into a dataframe
+        input_sentences_df = pd.DataFrame(
+            input_sentences,
+            columns=[
+                "sentence_text_id",
+                "corrected_text",
+                "input_language",
+                "context",
+                "quality_status",
+                "metadata",
+            ],
+        )
+
+        # Keep only the clean sentences
+        input_sentences_df = input_sentences_df[
+            (input_sentences_df["quality_status"] == "Clean")
+        ].reset_index(drop=True)
+
+        # Check if the dataframe is empty
+        if input_sentences_df.shape[0] == 0:
+            return {"error": "No clean sentences found. Perform project export first."}
+
+        # Make a sentence list for valid sentences to be translated
+        all_sentences_to_be_translated = input_sentences_df["corrected_text"].tolist()
+
+        # Loop through all the sentences to be translated in batch format
+        for i in range(0, len(all_sentences_to_be_translated), batch_size):
+
+            batch_of_input_sentences = all_sentences_to_be_translated[
+                i : i + batch_size
+            ]
+
+            # Check the API type
+            if api_type == "indic-trans":
+
+                # Get the translation using the Indictrans NMT API
+                translations_output = get_batch_translations_using_indictrans_nmt_api(
+                    batch_of_input_sentences,
+                    input_sentences_df["input_language"].iloc[0],
+                    output_language,
+                )
+
+                # Check if translation output didn't return an error
+                if isinstance(translations_output, Exception):
+                    return translations_output
+
+                # Collect the translated sentences
+                translated_sentences = [
+                    translation["target"] for translation in translations_output
+                ]
+
```

Comment:
Since we might add other services as well, to keep this function generic, shall we just move this inside `get_batch_translations_using_indictrans_nmt_api()` itself, and ensure that it returns an array of direct translations?

Reply:
Do you want to do this for the Google Translate API as well? I wanted to keep the processing inside the api functions as minimum as possible but I can do it for both the functions then.

Reply:
Yes please, it's better to keep API specific logic inside API specific utilities, and keep this function generic.